<h3><b>Significância Estatística e ANOVA na Análise de Dados.</h3><b>

<b>Vemos Padrões Em Todo Lugar</b>

O cérebro humano evoluiu para encontrar padões, uma habilidade vital para a sobrevivencia. Mas na análise de dados, essa habilidade pode nos enganar, fazendo-nos ver conexões onde só existe o acaso. A significância estatistica é o nosso filtro, é a ferramenta que separa o sinal do ruído e garante que nossas decisões sejam evidências, não coincidências.

<b>Pilares da Significancia Estatística</b>

<b>O Paradoxo do Valor-p</b>

O valor-p não mede a magnitude de um efeito. Ele reponde a uma pergunta especifica:<i> Se a hipótese nula $H_0$ for verdadeira, qual a probabilidade de obtermos estes dados (ou mais extremos) apenas por acaso?"</i>

Um p-valor de 0.05 não é um "limite mágico". É um equivoco comum pensar que um p-valor muito pequeno significa um efeito grande ou significativo. O p-valor indica apenas a probabilidade de obter um resultado observado sob a hipótese nula. Ele não fornece informações sobre a dimensão ou o significado prático do efeito.

<b>

<b>Poder Estatístico (1-$\beta$)</b>

O poder de um teste (1-$\beta$), também conhecido como a sensibilidade do teste, é a probabilidade de que um teste estatistico rejeite a hipotese nula ($H_0$) quando ela é falsa. Em termos mais simples, é a capacidade do seu teste de detectar um efeito ou uma diferença real que existe na população. Quanto maior o poder do teste, maior a sua chance de identificar os "verdadeiros positivos" e evitar um Erro do Tipo II, que ocorreria ao não detectar um efeito real.

<b>Fatores Criticos que Aumentam o Poder</b>

<ul>
    <li><b>Tamanho Amostral:</b> Amostras maiores geralmentem resultam em maior poder. Isso ocorre porque aamostras maiores fornecem mais informações sobre a população, reduzindo a incerteza e tornando mais facil identificar diferenças ou efeitos que realmente existem.
    <ul>
    <li><b>Ex.:</b> Imagina que você está testando uma nova funcionalidade em um site. Se você testar apenas 10 usuários, uma pequena, mas real, melhoria na taxa de conversão pode ser facilmente mascarada por variações aleatórias. No entanto, se você testar 1000 usuários, a verdadeira melhoria se tornará mais evidente, aumentando a probabilidade de o teste detecta-la como estatisticamente significativa.
    </ul>
    <li><b>Tamanho do efeito:</b> O tamanho do efeito refere-se à magnitude da diferença ou da relação que você está tentando detectar. Um tamanho de efeito maior leva a resultados estatisticamente mais significativos mais rápidamente, mesmo com grupos menores. Isso porque um efeito mais pronunciado é um "sinal" mais forte nos dados, tornando-o mais fácil de ser distiguido do "ruido" aleatório.
    <ul>
    <li><b>Ex.:</b> Se um medicamento reduz a pressão arterial em 20 mmHg, essa é uma diferença muito mais fácil de detectar estatisticamente do que uma redução de apenas 1 mmHg, mesmo com o mesmo númeor de pacientes. O efeito de 20 mmHg é um "sinal" muito mais claro.
    </ul>
    <li><b>Nível de Significância: </b> O nível de significância é a probabilidade predefinida de cometer um Erro do Tipo I (falso positivo), ou seja, rejeitar a hipotese nula quando ela é verdadeira. Um erro de significancia mais alto (Por exemplo 0.10 em vez de 0.05) leva a um maior poder do teste, pois expande a região de rejeição da H0, tornando mais provavel que você declare um resultado como significativo. No entanto, isso vem com o custo de aumentas o risco de Erro do Tipo I.
    <ul>
    <li><b>Ex.:</b> Se você define seu α em 0.10 (10% de chance de um falso positivo), você está mais propenso a encontrar um resultado "significativo" do que se o α fosse 0.01 (1% de chance). Isso aumenta sua capacidade de detectar um efeito real (maior poder), mas também aumenta o risco de concluir que há um efeito quando, na verdade, não há
    </ul>
</ul>

<b>Quando os Grupos se Multiplicam: Entra a ANOVA</b>

A Análise de Variância (ANOVA) é uma ferramenta utilizada para comparar as médias de três ou mais grupos simultaneamente. Seu principal objetivo é determinar se a variação observada entre as médias dos grupos é mais significativa que a variação dentro dos própios grupos. 
O Uso de multiplos teste t para cada par de grupos aumentaria o risco de Erro Tipo I (falso positivo), enquanto a ANOVA lida com a comparação em um unico teste e controla esse erro.

<b>Exemplo:</b>
<ul>
    <li>Eficácia de 4 medicamentos diferentes (comparar a eficácia de três métodos de ensaio).
    <li>Desempenho de vandas em 5 regiões (avaliar o crescimento médio de vendas para diferentes estratégias de marketing).
    <li>Satisfação de clientes em 3 canais de atendimento (identificar tendências em atendimento ao cliente para garantia da qualidade).
</ul>

A ANOVA compara duas estimativas de variância:
<ul>
    <li><b>Variância entre grupos:</b> Reflete as diferenças entre as médias dos diferentes grupos (causada por um fator especifico).
    <li><b>Variância intra-grupos:</b> Reflete as flutuações aleatórias e é causada por diferenças dentro de grupos individuais.
</ul>

Razão F 
$$F = \frac{\text{Variância Entre Grupos}}{\text{Variância Dentro de Grupos}}$$

Se a variação entre os grupos for maior significativamente maior que a variação dentro deles, concluímos que pelo menos um grupo é diferente.


Para garantir a validade dos resultados da ANOVA, certas premissas devem ser atendidas:
<ul>
    <li><b>Normalidade:</b> Os dados de cada grupo devem seguir uma distribuição normal.
    <li><b>Homocedasticidade:</b> As variações dentro de cada grupos devem ser aproximadamente iguais (homogeneidade de variâncias), o que pode ser testado com o teste de Levene.
    <li><b>Independência:</b> As observações em um grupo não devem influenciar as observações em outro grupo.
</ul>

Caso alguma dessas premissas seja violada, os resultados do teste podem ser inválidos, e o uso de um teste paramétrico pode ser necessário.

<b>Tipos de ANOVA</b>

| Tipo | Variáveis Independentes | Interações | Caso de Uso |
|------|-------------------------|------------|-------------|
| One-Way | 1 | Não | Comparar a efecácia de 3 métodos de ensino. |
| Two-Ways | 2 | Sim | Intestigar a relação entre ensino e técnicas de estudo, e como eles afetam conjuntamente o desempenho do aluno. |
| MANOVA | 1 ou mais | Sim | Comparar médias de multiplos grupos em duas ou mais variáveis dependentes (Ex.: impacto de diferentes configurações de asa em força estrutural, peso e eficacia aerodinâmica.)


<b>ANOVA One-Way com Dados Simulados</b>

In [2]:
import scipy.stats as stats
import pandas as pd
import numpy as np

residuos = [np.random.normal(0, 1, 100) for _ in range(3)]
grupo_A = np.random.normal(10, 2, 50)
grupo_B = np.random.normal(12, 2.5, 50)
grupo_C = np.random.normal(11, 2.2, 50)


_, p_normalidade = stats.shapiro(residuos)
print(f'Normalidade: {'OK' if p_normalidade > 0.05 else 'Violada'}')


data_normal = stats.norm.rvs(loc=0, scale=1, size=10)
_, p_normalidade = stats.shapiro(data_normal)
print(f'Normalidede (Exemplo): {'OK' if p_normalidade > 0.05 else 'Violada'}')

grupo_A = stats.norm.rvs(loc=10, scale=2, size=30)
grupo_B = stats.norm.rvs(loc=12, scale=2.5, size=30)
grupo_C = stats.norm.rvs(loc=11, scale=2.2, size=30)
_, p_homogeneidade = stats.levene(grupo_A, grupo_B, grupo_C) # [14, 15]
print(f"Homogeneidade (exemplo): {'OK' if p_homogeneidade > 0.05 else 'Violada'}")

Normalidade: OK
Normalidede (Exemplo): Violada
Homogeneidade (exemplo): Violada


In [3]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import pandas as pd
import numpy as np

# Dados de exemplo
np.random.seed(42)
data = {
    'grupo': ['A']*10 + ['B']*10 + ['C']*10, 
    'desempenho': np.concatenate([
        np.random.normal(70, 5, 10), # Média 70 para Grupo A
        np.random.normal(75, 5, 10), # Média 75 para Grupo B
        np.random.normal(68, 5, 10)  # Média 68 para Grupo C
    ])
}
dados = pd.DataFrame(data)

# Modelo ANOVA One-Way
# 'desempenho ~ C(grupo)' indica que 'desempenho' é a variável dependente
# e 'grupo' é a variável independente categórica.
modelo = ols('desempenho ~ C(grupo)', data=dados).fit()
modelo_anova = sm.stats.anova_lm(modelo, typ=2) # typ=2 para tipo de soma de quadrados
print("ANOVA One-Way:")
print(modelo_anova) # [16]

# Teste Post-Hoc (Tukey HSD)
# Compara pares de grupos para identificar quais diferenças são significativas
tukey = pairwise_tukeyhsd(
    endog=dados['desempenho'],
    groups=dados['grupo'],
    alpha=0.05
)
print("\nTeste Post-Hoc (Tukey HSD):")
print(tukey.summary())

ANOVA One-Way:
              sum_sq    df         F    PR(>F)
C(grupo)  157.712666   2.0  5.389779  0.010726
Residual  395.029386  27.0       NaN       NaN

Teste Post-Hoc (Tukey HSD):
Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj   lower   upper  reject
----------------------------------------------------
     A      B  -1.1936 0.7668 -5.4349  3.0477  False
     A      C  -5.3495 0.0113 -9.5908 -1.1082   True
     B      C  -4.1559 0.0556 -8.3972  0.0854  False
----------------------------------------------------


<b>Interpretação dos Resultados</b>

<ul>
<li><b>(Linha de Fator):</b>
<ul>
    <li><b>sum_sq (Soma dos Quadrados):</b> 157.712666 representa a variação entre os grupos. É a parte da variabilidade total dos dados que pode ser explicada pelas diferenças entre as médias dos seus métodos de estudos (A, B, C).
    <li><b>df (Graus de Liberdade):</b> 2.0 são os graus de liberdade para obter o fator grupo. É o número de grupos menor 1 (3 grupos -1 = 2).
    <li><b>F (Estatistica F):</b> é o valor da estatistica F calculada. É a razão entre a variação entre os grupos e a variação dentro dos grupos. Um fator F maior sugere que as diferenças entre as médias dos grupos são mais significativas do que a aleatóriedade. 
    <li><b>PR(>F) (p-valor): </b>0.010726 é o p-valor associado à estatistica F. Esse é o número mais importante nesta tabela. Nós comparamos esse p-valor com nosso nível de significancia, que geralmente é 0.05. Como 0.010726 é menor que 0.05, nós rejeitamos a hipótese nula da ANOVA.
</ul>
<li><b>Residual (linha do Residuo/Erro):</b>
<ul>
    <li><b>sum_sq:</b> 395.029386 representa a variação dentro dos grupos (ou variaveis do erro). É a parte da variabilidade que não é explicada pelas disferenças entre os grupos e é atribuida ao acaso ou fatores não medidos.
    <li><b>df:</b> 27.0 são os graus d eliberdade para o resíduo (número total de obeservações - número de grupos = 30 - 3 = 27)
</ul>

<b>Conclusão da ANOVA:</b>

O p-valor de 0.010726 (que é menor que 0.05) no diz que existe uma diferença estatisticamente significativa entre as notas médias de pelo menos um dos metodos de estudo.

No entanto, a ANOVA não nos diz quais grupos são diferentes. Para isso, precisamos dos testes Post-Hoc

<b>Interpretando os Resultados do Teste Post-Hoc (Tukey HSD)</b>

O teste de Tukey HSD é feito após a ANOVA ser significativa para identificar quais pares de grupos são responsáveis pela diferença observada. Ele compara cada par de grupo.

In [4]:
print("\nTeste Post-Hoc (Tukey HSD):")
print(tukey.summary())


Teste Post-Hoc (Tukey HSD):
Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj   lower   upper  reject
----------------------------------------------------
     A      B  -1.1936 0.7668 -5.4349  3.0477  False
     A      C  -5.3495 0.0113 -9.5908 -1.1082   True
     B      C  -4.1559 0.0556 -8.3972  0.0854  False
----------------------------------------------------


Aqui, o foco é na coluna <b>p-adj (p-valor ajustado)</b> e na coluna <b>breject</b>.
<ul>
    <li><b>group1</b> e <b>group2</b>: São os pares de grupos que estão sendo comparados.
    <li><b>meandif</b>: A diferença nas medias entre os grupos comparados. Por exemplo, <b>A</b> tem um dempenho 1.1936 pontos menor que <b>B</b>, e <b>A</b> tem um desempenho 5.3495 pontos menor que <b>C</b>.
    <li><b>p-adj (p-valor ajustado)</b>: Esse é o valor para cada comparação pareado, ajustado para o problema da testagem multipla (para evitar falsos positivos quando se faz várias comparações.)
    <b>upper</b> e <b>lower</b>: são os limites inferiores e superiores do intervalor de confiança para a diferença nas médias.
    <b>reject</b>: Uma coluna booleana que indica se a hipótese nula para aquele par especifico for rejeitada.
</ul>

<b>Analisando as Comparações</b>

1. <b>A vs. B</b>
<ul>
<li>meandiff = -1.1936 (tem a média ligeiramente menor)
<li>p-adj = 0.7668
<li>reject = False
<li><b>Interpretação</b>: O p-valor ajustado (0.7668) é muito maior que o 0.05. Não há diferença estatisticamente significativa entre as médias dos grupos A e B.
</ul>

2. <b>A vs. C</b>
<ul>
<li>meandiff = -5.3495 (A tem a média significativamente menor que C)
<li>p-adj = 0.0113
<li>reject = True
<li><b>Interpreatação</b>: O p-valor ajustado (0.0113) é menor que 0.05. Há diferença estatisticamente siginificativamente entre as médias dos grupos A e C. O grupo C teve um desempenho siginificativamente melhor que o grupo A.
</ul>

3. <b>B vs. C</b>
<ul>
<li>meandiff = -4.1559 (B tem média menor que C)
<li>p-adj = 0.0556
<li>reject = False
<li><b>Interpretação</b>: O p-valor ajustado (0.0556) é ligeiramente maior que 0.05. Portanto, não há diferença estatisticamente significativa entre as médias dos grupos B e C ao nível de 0.05. Note que o p-valor está bem na fronteira, segurindo que talvez haja uma diferença, mas não suficiente para ser considerada "significativa" pelo critério de 0.05.
</ul>

<b>Conclusão Geral</b>
Com base nos resultados:
<ul>
<li>A ANOVA nos disse que exite uma diferença geral nas médias de desempenho entre métodos de estudos.
<li>O Teste de Post-Hoc Tukey nos revelou que a unica diferença estatisticamente significativa (ao nível de alpha = 0.05) está entre o grupo A e o grupo C, com o grupo C apresentando um desempenho significativamente superior.
<li>Não há evidencias de diferenças significativas entre A e B, ou entre B e C, embora a comparação entre B vs. C esteja muito próxima do limiar. 
</ul>

Isso significa que, com base nos dados simulados, o "Metodo C" parece ter levado a resultados melhores do que o metodo "A".

<b>Conclusão</b>

Dominar significância estatística e ANOVA transforma analistas em parceiros estratégicos. A significância estatística é uma ferramenta indispensável para a análise de dados, capacitando as empresas a tomar decisões sólidas e fundamentadas em evidências, otimizando assim o retorno sobre o investimento.9 Lembre-se:
"Estatística é a ciência de evitar ser enganado por dados aleatórios." (Brad Efron)
